rust/1_38_0.nix

```diff
@@ -1,22 +0,0 @@
-import ./default.nix {
```

Comment:
The only place were those files were referred to was through there:
```
// pkgs.lib.optionalAttrs (nixpkgs == "nixpkgs-20.03" && pkgs.stdenv.isLinux) muslTests
```
... but there's no mention of `nixpkgs-20.03` anywhere else, so those `rust/*` files (required for some of the `muslTests`) were effectively not used anyway.

Instead of rewriting them to use a newer nixpkgs, I've decided that it's probably best if we get rid of them and later create a separate wasm/musl/etc. testsuite that doesn't require compiling rustc from scratch (e.g. by using nixpkgs-mozilla).

Reply:
Ah, I see! Then this was definitely overlooked when we upgraded from `nixpkgs-20.03`. I think there was one particular use case for those tests at $WORK but I don't mind if we simply get rid of them. Either way let's not add a dependency on nixpkgs-mozilla.